### ISSUE TITLE
VB warning BC40003 "use the overloads modifier" should have a quickfix that inserts that modifier

### ISSUE THREAD

**Issue by ljw1004** [dotnet/roslyn#3437]
Look at this screenshot:
![bc40003](https://cloud.githubusercontent.com/assets/3316258/8099226/8a51a3b8-0fae-11e5-8742-f0e5a5535601.png)

The warning message BC40003 says "property 'Current' shadows an overloadable member declared in the base class 'Application'.  **If you want to overload the base method, this method must be declared 'Overloads'.**"

WHAT I EXPECT: there should be a quickfix offering to add the `Overloads` modifier

WHAT I GET: it only offers to suppress this message

**Comment by Pilchie** [comment]
This is something that could be very easily contributed.

**Comment by Giftednewt** [comment]
I'd like to contribute the codefixprovider for this warning. I've started looking into it but am not able to debug the code changes I have made. Should it be possible to debug the changes made in this layer?

**Comment by Pilchie** [comment]
Sadly, not yet :sob:

The VS Extensibility  team is working on supporting this, but it won't be until at least VS 2015 Update 1.

In the meantime, your choices are:
1. Share the code into a separate VSIX project and verify that it works from that VSIX.
2. Rely on unit tests that you can run, and ask one of us to verify the end to end experience once you have a PR in place.

**Comment by Giftednewt** [comment]
That's a shame. I went with option 1 as it seemed a more reliable way of verifying that the provider worked even though the unit tests were passing.

I've got the provider adding the Overloads modifier in but at present it's creating the following
```VB
Shared Overloads Property Current As App
```
But this is not correct as once I move the caret off the property line the modifiers get re-ordered giving the following
```VB
Overloads Shared Property Current As App
```
Is there a service that can be used from within the BasicFeatures project to re-order the modifiers when I create the new node based on the VB rules?

**Comment by Giftednewt** [comment]
@Pilchie I see that this is also an issue  in C# being reported as CS0108. If you're in agreement I'll create something to fix that as well while I'm at it?

With regards to my previous comment I see that modifier reordering is implemented in a code cleanup provider so didn't want to duplicate the logic. Is there a way that I can access that as part of the CodeFixAction?

**Comment by Pilchie** [comment]
Hmm - we do this in formatting using [`NormalizeModifiersOrOperatorsCodeCleanupProvider`](http://source.roslyn.io/Microsoft.CodeAnalysis.VisualBasic.Workspaces/R/4888643484c9129c.html), but we don't seem to offer a public way to access that :frowning:

**Comment by paul1956** [comment]
I'll admit I don't really understand "open source", if I need this functionality and you provide the source can I just copy the source into my Fix Provider or do I need to reinvent the wheel?

**Comment by Pilchie** [comment]
You should take a look at the [License](https://github.com/dotnet/roslyn/blob/master/License.txt) terms, which document all of this.

**Comment by Giftednewt** [comment]
It looks like I'm able to access that provider through a call to the GetLanguageService extension method against the document retrieving an instance of ICodeCleanerService. Once I have that I can call Clean up passing in the SyntaxNode and I get the correct result.

It's not particularly pretty but it does solve the problem of not being able to access NormalizeModifiersOrOperatorsCodeCleanupProvider directly.

**Comment by Pilchie** [comment]
Oh, great.  That's actually the designed way to do it (I forgot that we made `ICodeCleanerService` public).  The reasoning is that this way you can write some language agnostic code for fixes where it makes sense, and then "clean them up" appropriately for either language.

**Comment by Youssef1313** [comment]
This was already fixed in #5299

@jinujoseph Can you close this?

**Comment by CyrusNajmabadi** [comment]
Confirmed.  Fixed with https://github.com/dotnet/roslyn/pull/5299.  Thanks @Youssef1313  :)
